src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1694,23 +1704,25 @@ private Matches() {}
     });
   }
 
-  public static final Match<Unit> UnitCanGiveBonusMovement = Match.of(unit -> {
-    final UnitAttachment ua = UnitAttachment.get(unit.getType());
-    if (ua == null) {
-      return false;
-    }
-    return ua.getGivesMovement().size() > 0 && Matches.unitIsBeingTransported().invert().match(unit);
-  });
+  private static Match<Unit> unitCanGiveBonusMovement() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
```

Comment:
This check could be merged with the return statement: `return ua != null && ua.getGivesMovement().size() > 0 && Matches.unitIsBeingTransported().invert().match(unit);`